internal/client/work_queues.go

```diff
@@ -0,0 +1,194 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/prefecthq/terraform-provider-prefect/internal/api"
+)
+
+var _ = api.WorkQueuesClient(&WorkQueuesClient{})
+
+// WorkQueuesClient is a client for working with work queues.
+type WorkQueuesClient struct {
+	hc          *http.Client
+	apiKey      string
+	routePrefix string
+}
+
+// WorkQueues returns a WorkQueuesClient.
+//
+//nolint:ireturn // required to support PrefectClient mocking
+func (c *Client) WorkQueues(accountID uuid.UUID, workspaceID uuid.UUID, workPoolName string) (api.WorkQueuesClient, error) {
+	if accountID == uuid.Nil {
+		accountID = c.defaultAccountID
+	}
+	if workspaceID == uuid.Nil {
+		workspaceID = c.defaultWorkspaceID
+	}
+
+	var workspaceScopedURL = getWorkspaceScopedURL(c.endpoint, accountID, workspaceID, "work_pools")
+
+	var workPoolScopedURL = workspaceScopedURL + "/" + workPoolName + "/queues"
```

Comment:
not opposed to this, but that means one of these `WorkQueue` client instances will be tied to a specific workPool